src/common/util/include/openvino/util/node_util.hpp

```diff
@@ -0,0 +1,27 @@
+// Copyright (C) 2018-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include "openvino/core/node.hpp"
+
+namespace ov {
+namespace util {
+/**
+ * @brief Set name for both node and output tensor. Any other names will be overriden by a given single name
+ * @param node - node to rename
+ * @param name - new name
+ * @param output_idx - idx of output to rename
+ * @return - renamed node
+ */
+std::shared_ptr<ov::Node> set_name(std::shared_ptr<ov::Node> node, const std::string& name, size_t output_idx = 0);
```

Comment:
Consider rename, as this function sets friendly name and output port name, will be not obvious what is set.

If functions only set name the signature could be:
```suggestion
OPENVINO_API void set_name(ov::Node& node, const std::string& name, size_t output_port = 0);
```
There is no need to pass shared pointer, as nothing is done with ownership.

What if Node has many outputs which on should be used or all?

Consider add some test where special nodes like Parameter/Result/Sink are tested.
Test node with single output and multiple outputs.

Reply:
done